Rx-java support

I want to put my credit card creation inside an observable for testing purposes: and I get:

```
java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.Worker thread.
            at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:62)
            at android.os.Handler.handleCallback(Handler.java:733)
            at android.os.Handler.dispatchMessage(Handler.java:95)
            at android.os.Looper.loop(Looper.java:146)
            at android.app.ActivityThread.main(ActivityThread.java:5694)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:515)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1291)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1107)
            at dalvik.system.NativeStart.main(Native Method)
     Caused by: java.lang.ExceptionInInitializerError
            at com.stripe.android.Stripe$1.create(Stripe.java:22)
            at com.stripe.android.Stripe.createToken(Stripe.java:144)
            at com.stripe.android.Stripe.createToken(Stripe.java:127)
            at com.stripe.android.Stripe.createToken(Stripe.java:131)
            at com.pickupnow.customer.model.payment.PaymentService.lambda$createToken$59(PaymentService.java:30)
            at com.pickupnow.customer.model.payment.PaymentService.access$lambda$0(PaymentService.java)
            at com.pickupnow.customer.model.payment.PaymentService$$Lambda$1.call(Unknown Source)
            at rx.Observable$1.call(Observable.java:144)
            at rx.Observable$1.call(Observable.java:136)
            at rx.Observable$1.call(Observable.java:144)
            at rx.Observable$1.call(Observable.java:136)
            at rx.Observable$1.call(Observable.java:144)
            at rx.Observable$1.call(Observable.java:136)
            at rx.Observable.unsafeSubscribe(Observable.java:7531)
            at rx.internal.operators.OperatorSubscribeOn$1$1.call(OperatorSubscribeOn.java:62)
            at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
            at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
            at java.util.concurrent.FutureTask.run(FutureTask.java:237)
            at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
            at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)
     Caused by: java.lang.RuntimeException: Can't create handler inside thread that has not called Looper.prepare()
            at android.os.Handler.<init>(Handler.java:200)
            at android.os.Handler.<init>(Handler.java:114)
            at com.stripe.android.compat.AsyncTask$InternalHandler.<init>(AsyncTask.java:579)
            at com.stripe.android.compat.AsyncTask$InternalHandler.<init>(AsyncTask.java:579)
            at com.stripe.android.compat.AsyncTask.<clinit>(AsyncTask.java:189)
            at com.stripe.android.Stripe$1.create(Stripe.java:22)
            at com.stripe.android.Stripe.createToken(Stripe.java:144)
            at com.stripe.android.Stripe.createToken(Stripe.java:127)
            at com.stripe.android.Stripe.createToken(Stripe.java:131)
            at com.pickupnow.customer.model.payment.PaymentService.lambda$createToken$59(PaymentService.java:30)
            at com.pickupnow.customer.model.payment.PaymentService.access$lambda$0(PaymentService.java)
            at com.pickupnow.customer.model.payment.PaymentService$$Lambda$1.call(Unknown Source)
            at rx.Observable$1.call(Observable.java:144)
            at rx.Observable$1.call(Observable.java:136)
            at rx.Observable$1.call(Observable.java:144)
            at rx.Observable$1.call(Observable.java:136)
            at rx.Observable$1.call(Observable.java:144)
            at rx.Observable$1.call(Observable.java:136)
            at rx.Observable.unsafeSubscribe(Observable.java:7531)
            at rx.internal.operators.OperatorSubscribeOn$1$1.call(OperatorSubscribeOn.java:62)
            at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
            at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
            at java.util.concurrent.FutureTask.run(FutureTask.java:237)
            at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
            at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)
```

When I execute:

```
public Observable<String> createToken(String number, int month, int year, String cvc, String zip) {
        return Observable.create(subscriber -> {
            Card card = new Card(number, month, year, cvc);
            card.setAddressZip(zip);

            if (card.validateCard()) {
                try {
                    Stripe stripe = new Stripe(BuildConfig.STRIPE_TOKEN);

                    stripe.createToken(card, new TokenCallback() {
                        @Override
                        public void onSuccess(final Token token) {
                            subscriber.onNext(token.getId());
                            subscriber.onCompleted();
                        }

                        @Override
                        public void onError(Exception e) {
                            subscriber.onError(e);
                        }
                    });
                } catch (AuthenticationException e) {
                    subscriber.onError(e);
                }
            } else {
                subscriber.onError(new InvalidCardException("Invalid card values"));
            }
        });
    }
```

--- a/example/build.gradle
+++ b/example/build.gradle
@@ -6,8 +6,13 @@ repositories {
 
 dependencies {
     compile project(':stripe')
-    compile 'com.android.support:appcompat-v7:22.2.1'
-    compile 'com.android.support:support-v4:24.1.1'
+    compile 'com.android.support:appcompat-v7:25.0.1'
+    compile 'com.android.support:support-v4:25.0.1'
+    /* Needed for RxAndroid */
+    compile 'io.reactivex:rxandroid:1.2.1'
+    compile 'io.reactivex:rxjava:1.1.6'
+    /* Needed for Rx Bindings on views */
+    compile 'com.jakewharton.rxbinding:rxbinding:0.4.0'
 }
 
 android {

--- a/example/res/layout/payment_activity.xml
+++ b/example/res/layout/payment_activity.xml
@@ -76,6 +76,11 @@
         android:text="@string/save"
         style="@style/Save"/>
 
+    <Button
+        android:id="@+id/saverx"
+        android:text="@string/saverx"
+        style="@style/Save"/>
+
     <TextView android:id="@+id/token_list_title"
               android:text="@string/paymentMethods"
               android:layout_marginTop="12dp"

--- a/example/res/values/strings.xml
+++ b/example/res/values/strings.xml
@@ -11,6 +11,7 @@
 
     <string name="cardNumber">Card Number</string>
     <string name="save">Save</string>
+    <string name="saverx">Save Rx</string>
     <string name="cvc">CVC</string>
 
     <string name="ok">OK</string>

--- a/example/src/main/java/com/stripe/example/activity/PaymentActivity.java
+++ b/example/src/main/java/com/stripe/example/activity/PaymentActivity.java
@@ -95,6 +95,9 @@ public class PaymentActivity extends AppCompatActivity {
     @Override
     protected void onDestroy() {
         super.onDestroy();
+        if (compositeSubscription != null) {
+            compositeSubscription.unsubscribe();
+        }
     }
 
     private void addToList(Token token) {

--- a/example/src/main/java/com/stripe/example/activity/PaymentActivity.java
+++ b/example/src/main/java/com/stripe/example/activity/PaymentActivity.java
@@ -55,7 +55,7 @@ public class PaymentActivity extends AppCompatActivity {
     private SimpleAdapter simpleAdapter;
     private List<Map<String, String>> cardTokens = new ArrayList<Map<String, String>>();
 
-    // A CompositeSubscription object to make cleaning up simpler
+    // A CompositeSubscription object to make cleaning up our rx subscriptions simpler
     private CompositeSubscription compositeSubscription;
 
     @Override
@@ -97,7 +97,89 @@ public class PaymentActivity extends AppCompatActivity {
         super.onDestroy();
     }
 
-    public void saveCreditCard() {
+    private void addToList(Token token) {
+        String endingIn = getResources().getString(R.string.endingIn);
+        Map<String, String> map = new HashMap<>();
+        map.put("last4", endingIn + " " + token.getCard().getLast4());
+        map.put("tokenId", token.getId());
+        cardTokens.add(map);
+        simpleAdapter.notifyDataSetChanged();
+    }
+
+    private Card createCardToSave() {
+        String cardNumber = cardNumberEditText.getText().toString();
+        String cvc = cvcEditText.getText().toString();
+
+        int expMonth = getIntegerFromSpinner(monthSpinner);
+        int expYear = getIntegerFromSpinner(yearSpinner);
+
+        String currency = getCurrency();
+        Card cardToSave = new Card(cardNumber, expMonth, expYear, cvc);
+        cardToSave.setCurrency(currency);
+        return cardToSave;
+    }
+
+    private boolean validateCard(Card card) {
+        boolean valid = true;
+        if (!card.validateNumber()) {
+            handleError("The card number that you entered is invalid");
+            valid = false;
+        } else if (!card.validateExpiryDate()) {
+            handleError("The expiration date that you entered is invalid");
+            valid = false;
+        } else if (!card.validateCVC()) {
+            handleError("The CVC code that you entered is invalid");
+            valid = false;
+        } else if (!card.validateCard()){
+            handleError("The card details that you entered are invalid");
+            valid = false;
+        }
+
+        return valid;
+    }
+
+    private void finishProgress() {
+        progressFragment.dismiss();
+    }
+
+    private String getCurrency() {
+        if (currencySpinner.getSelectedItemPosition() == 0) {
+            return null;
+        }
+
+        String selected = (String) currencySpinner.getSelectedItem();
+
+        if (selected.equals(CURRENCY_UNSPECIFIED)) {
+            return null;
+        }
+
+        return selected.toLowerCase();
+    }
+
+    private int getIntegerFromSpinner(Spinner spinner) {
+        try {
+            return Integer.parseInt(spinner.getSelectedItem().toString());
+        } catch (NumberFormatException e) {
+            return 0;
+        }
+    }
+
+    private void handleError(String error) {
+        DialogFragment fragment = ErrorDialogFragment.newInstance(R.string.validationErrors, error);
+        fragment.show(getSupportFragmentManager(), "error");
+    }
+
+    private void initListView() {
+        simpleAdapter = new SimpleAdapter(
+                this,
+                cardTokens,
+                R.layout.list_item_layout,
+                new String[]{"last4", "tokenId"},
+                new int[]{R.id.last4, R.id.tokenId});
+        listView.setAdapter(simpleAdapter);
+    }
+
+    private void saveCreditCard() {
         Card cardToSave = createCardToSave();
 
         if (validateCard(cardToSave)) {
@@ -155,89 +237,7 @@ public class PaymentActivity extends AppCompatActivity {
                         }));
     }
 
-    private void addToList(Token token) {
-        String endingIn = getResources().getString(R.string.endingIn);
-        Map<String, String> map = new HashMap<>();
-        map.put("last4", endingIn + " " + token.getCard().getLast4());
-        map.put("tokenId", token.getId());
-        cardTokens.add(map);
-        simpleAdapter.notifyDataSetChanged();
-    }
-
-    private Card createCardToSave() {
-        String cardNumber = cardNumberEditText.getText().toString();
-        String cvc = cvcEditText.getText().toString();
-
-        int expMonth = getIntegerFromSpinner(monthSpinner);
-        int expYear = getIntegerFromSpinner(yearSpinner);
-
-        String currency = getCurrency();
-        Card cardToSave = new Card(cardNumber, expMonth, expYear, cvc);
-        cardToSave.setCurrency(currency);
-        return cardToSave;
-    }
-
-    private boolean validateCard(Card card) {
-        boolean valid = true;
-        if (!card.validateNumber()) {
-            handleError("The card number that you entered is invalid");
-            valid = false;
-        } else if (!card.validateExpiryDate()) {
-            handleError("The expiration date that you entered is invalid");
-            valid = false;
-        } else if (!card.validateCVC()) {
-            handleError("The CVC code that you entered is invalid");
-            valid = false;
-        } else if (!card.validateCard()){
-            handleError("The card details that you entered are invalid");
-            valid = false;
-        }
-
-        return valid;
-    }
-
     private void startProgress() {
         progressFragment.show(getSupportFragmentManager(), "progress");
-    }
-
-    private void finishProgress() {
-        progressFragment.dismiss();
-    }
-
-    private String getCurrency() {
-        if (currencySpinner.getSelectedItemPosition() == 0) {
-            return null;
-        }
-
-        String selected = (String) currencySpinner.getSelectedItem();
-
-        if (selected.equals(CURRENCY_UNSPECIFIED)) {
-            return null;
-        }
-
-        return selected.toLowerCase();
-    }
-
-    private int getIntegerFromSpinner(Spinner spinner) {
-        try {
-            return Integer.parseInt(spinner.getSelectedItem().toString());
-        } catch (NumberFormatException e) {
-            return 0;
-        }
-    }
-
-    private void handleError(String error) {
-        DialogFragment fragment = ErrorDialogFragment.newInstance(R.string.validationErrors, error);
-        fragment.show(getSupportFragmentManager(), "error");
-    }
-
-    private void initListView() {
-        simpleAdapter = new SimpleAdapter(
-                this,
-                cardTokens,
-                R.layout.list_item_layout,
-                new String[]{"last4", "tokenId"},
-                new int[]{R.id.last4, R.id.tokenId});
-        listView.setAdapter(simpleAdapter);
     }
 }

--- a/example/src/main/java/com/stripe/example/activity/PaymentActivity.java
+++ b/example/src/main/java/com/stripe/example/activity/PaymentActivity.java
@@ -10,6 +10,7 @@ import android.widget.ListView;
 import android.widget.SimpleAdapter;
 import android.widget.Spinner;
 
+import com.jakewharton.rxbinding.view.RxView;
 import com.stripe.android.Stripe;
 import com.stripe.android.TokenCallback;
 import com.stripe.android.model.Card;
@@ -22,6 +23,13 @@ import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.Callable;
+
+import rx.Observable;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.functions.Action1;
+import rx.schedulers.Schedulers;
+import rx.subscriptions.CompositeSubscription;
 
 public class PaymentActivity extends AppCompatActivity {
 
@@ -47,6 +55,8 @@ public class PaymentActivity extends AppCompatActivity {
     private SimpleAdapter simpleAdapter;
     private List<Map<String, String>> cardTokens = new ArrayList<Map<String, String>>();
 
+    // A CompositeSubscription object to make cleaning up simpler
+    private CompositeSubscription compositeSubscription;
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
@@ -63,6 +73,16 @@ public class PaymentActivity extends AppCompatActivity {
             }
         });
 
+        compositeSubscription = new CompositeSubscription();
+        Button saveRxButton = (Button) findViewById(R.id.saverx);
+        compositeSubscription.add(
+                RxView.clicks(saveRxButton).subscribe(new Action1<Void>() {
+                    @Override
+                    public void call(Void aVoid) {
+                        saveCreditCardWithRx();
+                    }
+                }));
+
         this.cardNumberEditText = (EditText) findViewById(R.id.number);
         this.cvcEditText = (EditText) findViewById(R.id.cvc);
         this.monthSpinner = (Spinner) findViewById(R.id.expMonth);
@@ -72,16 +92,13 @@ public class PaymentActivity extends AppCompatActivity {
         initListView();
     }
 
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+    }
+
     public void saveCreditCard() {
-        String cardNumber = cardNumberEditText.getText().toString();
-        String cvc = cvcEditText.getText().toString();
-
-        int expMonth = getIntegerFromSpinner(monthSpinner);
-        int expYear = getIntegerFromSpinner(yearSpinner);
-
-        String currency = getCurrency();
-        Card cardToSave = new Card(cardNumber, expMonth, expYear, cvc);
-        cardToSave.setCurrency(currency);
+        Card cardToSave = createCardToSave();
 
         if (validateCard(cardToSave)) {
             startProgress();
@@ -101,13 +118,63 @@ public class PaymentActivity extends AppCompatActivity {
         }
     }
 
+    private void saveCreditCardWithRx() {
+        final Card cardToSave = createCardToSave();
+        if (!validateCard(cardToSave)) {
+            return;
+        }
+
+        final Stripe stripe = new Stripe();
+        final Observable<Token> tokenObservable =
+                Observable.fromCallable(
+                        new Callable<Token>() {
+                            @Override
+                            public Token call() throws Exception {
+                                startProgress();
+                                return stripe.createTokenSynchronous(cardToSave, PUBLISHABLE_KEY);
+                            }
+                        });
+
+        compositeSubscription.add(tokenObservable
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        new Action1<Token>() {
+                            @Override
+                            public void call(Token token) {
+                                addToList(token);
+                                finishProgress();
+                            }
+                        },
+                        new Action1<Throwable>() {
+                            @Override
+                            public void call(Throwable throwable) {
+                                handleError(throwable.getLocalizedMessage());
+                                finishProgress();
+                            }
+                        }));
+    }
+
     private void addToList(Token token) {
         String endingIn = getResources().getString(R.string.endingIn);
-        Map<String, String> map = new HashMap<String, String>();
+        Map<String, String> map = new HashMap<>();
         map.put("last4", endingIn + " " + token.getCard().getLast4());
         map.put("tokenId", token.getId());
         cardTokens.add(map);
         simpleAdapter.notifyDataSetChanged();
+    }
+
+    private Card createCardToSave() {
+        String cardNumber = cardNumberEditText.getText().toString();
+        String cvc = cvcEditText.getText().toString();
+
+        int expMonth = getIntegerFromSpinner(monthSpinner);
+        int expYear = getIntegerFromSpinner(yearSpinner);
+
+        String currency = getCurrency();
+        Card cardToSave = new Card(cardNumber, expMonth, expYear, cvc);
+        cardToSave.setCurrency(currency);
+        return cardToSave;
     }
 
     private boolean validateCard(Card card) {

--- a/stripe/src/main/java/com/stripe/android/Stripe.java
+++ b/stripe/src/main/java/com/stripe/android/Stripe.java
@@ -431,9 +431,9 @@ public class Stripe {
         return tokenParams;
     }
 
-    public class ResponseWrapper {
-        public final Token token;
-        public final Exception error;
+    private class ResponseWrapper {
+        final Token token;
+        final Exception error;
 
         private ResponseWrapper(Token token, Exception error) {
             this.error = error;

--- a/stripe/src/main/java/com/stripe/android/Stripe.java
+++ b/stripe/src/main/java/com/stripe/android/Stripe.java
@@ -153,7 +153,7 @@ public class Stripe {
     /**
      * Call to create a {@link Token} with the publishable key and {@link Executor} specified.
      *
-     * @param card the {@link Card} used for this payment token
+     * @param card the {@link Card} used for this token
      * @param publishableKey the publishable key to use
      * @param executor an {@link Executor} to run this operation on. If null, this is run on a
      *                 default non-ui executor
@@ -185,6 +185,20 @@ public class Stripe {
         }
     }
 
+    /**
+     * Blocking method to create a {@link Token}. Do not call this on the UI thread or your app
+     * will crash. This method uses the default publishable key for this {@link Stripe} instance.
+     *
+     * @param card the {@link Card} to use for this token
+     * @return a {@link Token} that can be used for this card
+     *
+     * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+     * @throws InvalidRequestException your request has invalid parameters
+     * @throws APIConnectionException failure to connect to Stripe's API
+     * @throws CardException the card cannot be charged for some reason
+     * @throws APIException any other type of problem (for instance, a temporary issue with
+     * Stripe's servers
+     */
     public Token createTokenSynchronous(final Card card)
             throws AuthenticationException,
             InvalidRequestException,
@@ -194,8 +208,27 @@ public class Stripe {
         return createTokenSynchronous(card, defaultPublishableKey);
     }
 
+    /**
+     * Blocking method to create a {@link Token}. Do not call this on the UI thread or your app
+     * will crash.
+     *
+     * @param card the {@link Card} to use for this token
+     * @param publishableKey the publishable key to use with this request
+     * @return a {@link Token} that can be used for this card
+     *
+     * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+     * @throws InvalidRequestException your request has invalid parameters
+     * @throws APIConnectionException failure to connect to Stripe's API
+     * @throws CardException the card cannot be charged for some reason
+     * @throws APIException any other type of problem (for instance, a temporary issue with
+     * Stripe's servers
+     */
     public Token createTokenSynchronous(Card card, String publishableKey)
-            throws AuthenticationException, InvalidRequestException, APIConnectionException, CardException, APIException {
+            throws AuthenticationException,
+            InvalidRequestException,
+            APIConnectionException,
+            CardException,
+            APIException {
         validateKey(publishableKey);
 
         RequestOptions requestOptions = RequestOptions.builder()
@@ -205,22 +238,6 @@ public class Stripe {
         com.stripe.model.Card stripeCard = stripeToken.getCard();
         Card resultCard = androidCardFromStripeCard(stripeCard);
         return androidTokenFromStripeToken(resultCard, stripeToken);
-//        try {
-//            validateKey(publishableKey);
-//
-//            RequestOptions requestOptions = RequestOptions.builder()
-//                    .setApiKey(publishableKey).build();
-//            com.stripe.model.Token stripeToken = com.stripe.model.Token.create(
-//                    hashMapFromCard(card), requestOptions);
-//            com.stripe.model.Card stripeCard = stripeToken.getCard();
-//            Card resultCard = androidCardFromStripeCard(stripeCard);
-//            return androidTokenFromStripeToken(resultCard, stripeToken);
-////            return new ResponseWrapper(token, null);
-//        } catch (AuthenticationException authenticationException) {
-//            return new ResponseWrapper(null, authenticationException);
-//        } catch (Exception e) {
-//            return new ResponseWrapper(null, e);
-//        }
     }
 
     /**

--- a/stripe/src/main/java/com/stripe/android/Stripe.java
+++ b/stripe/src/main/java/com/stripe/android/Stripe.java
@@ -17,7 +17,11 @@ import java.util.concurrent.Executor;
 import com.stripe.android.model.Card;
 import com.stripe.android.model.Token;
 import com.stripe.android.util.StripeTextUtils;
+import com.stripe.exception.APIConnectionException;
+import com.stripe.exception.APIException;
 import com.stripe.exception.AuthenticationException;
+import com.stripe.exception.CardException;
+import com.stripe.exception.InvalidRequestException;
 import com.stripe.net.RequestOptions;
 
 /**
@@ -179,6 +183,44 @@ public class Stripe {
         catch (AuthenticationException e) {
             callback.onError(e);
         }
+    }
+
+    public Token createTokenSynchronous(final Card card)
+            throws AuthenticationException,
+            InvalidRequestException,
+            APIConnectionException,
+            CardException,
+            APIException {
+        return createTokenSynchronous(card, defaultPublishableKey);
+    }
+
+    public Token createTokenSynchronous(Card card, String publishableKey)
+            throws AuthenticationException, InvalidRequestException, APIConnectionException, CardException, APIException {
+        validateKey(publishableKey);
+
+        RequestOptions requestOptions = RequestOptions.builder()
+                .setApiKey(publishableKey).build();
+        com.stripe.model.Token stripeToken = com.stripe.model.Token.create(
+                hashMapFromCard(card), requestOptions);
+        com.stripe.model.Card stripeCard = stripeToken.getCard();
+        Card resultCard = androidCardFromStripeCard(stripeCard);
+        return androidTokenFromStripeToken(resultCard, stripeToken);
+//        try {
+//            validateKey(publishableKey);
+//
+//            RequestOptions requestOptions = RequestOptions.builder()
+//                    .setApiKey(publishableKey).build();
+//            com.stripe.model.Token stripeToken = com.stripe.model.Token.create(
+//                    hashMapFromCard(card), requestOptions);
+//            com.stripe.model.Card stripeCard = stripeToken.getCard();
+//            Card resultCard = androidCardFromStripeCard(stripeCard);
+//            return androidTokenFromStripeToken(resultCard, stripeToken);
+////            return new ResponseWrapper(token, null);
+//        } catch (AuthenticationException authenticationException) {
+//            return new ResponseWrapper(null, authenticationException);
+//        } catch (Exception e) {
+//            return new ResponseWrapper(null, e);
+//        }
     }
 
     /**
@@ -372,7 +414,7 @@ public class Stripe {
         return tokenParams;
     }
 
-    private class ResponseWrapper {
+    public class ResponseWrapper {
         public final Token token;
         public final Exception error;
 

